docs/howto/danceable-lyrics-input-tool/README.md

```diff
@@ -0,0 +1,485 @@
+
+```python
+self.provider.push_outgoing_metadata(
+    "Output",
+    create_schema(
+        {
+            "artist_name": {"type": FieldType.v_wstring,},
+            "track_name": {"type": FieldType.v_wstring,},
+            "danceability": {"type": FieldType.float,},
+            "energy": {"type": FieldType.float,},
+            "track_id": {"type": FieldType.v_wstring,},
+        }
+    ),
+)
+```
+
+Great! Now Designer and downstream tooling will know exactly what kind of data to expect from us!
+
+As we would like to find the "most danceable" songs based on particular lyrics, let's provide some terms or phrases. Our tool will gather songs (or "tracks") that contain lyrics with these terms.
+
+```python
+self.LYRICS_TERMS = [
+    "star wars",
+    "star trek",
+    "luke skywalker",
+    "captain kirk",
+    "spock",
+    "yoda",
+]
+```
+
+> :warning: Be aware that these datasets are *not* filtered in any way, and songs or titles may contain explicit content!
+
+Finally, let's add some additional criteria for our analysis and an output message to display when we are initialized. Feel free to play around with these values!
+
+```python
+self.DANCEABILITY_RANGE = [0.45, 0.99]
+self.ENERGY_RANGE = [0.45, 0.75]
+self.TEMPO_RANGE = [110.0, 140.0]
+self.MIN_VIEWS = 1000
+
+self.provider.io.info(f"{self.name} initialized.")
+```
+
+When you have completed the above steps, your `__init__` should look something like the following:
+
+```python
+def __init__(self, provider: AMPProviderV2) -> None:
+    """Construct a plugin."""
+    self.provider = provider
+    self.name = "DanceableLyrics"
+
+    # To build a more flexible plugin, read input path(s) from the user
+    self.DATASETS_BASE = "c:/users/bryan.ashby/DanceableLyricsData/"
+
+    self.provider.push_outgoing_metadata(
+        "Output",
+        create_schema(
+            {
+                "artist_name": {"type": FieldType.v_wstring,},
+                "track_name": {"type": FieldType.v_wstring,},
+                "danceability": {"type": FieldType.float,},
+                "energy": {"type": FieldType.float,},
+                "track_id": {"type": FieldType.v_wstring,},
+            }
+        ),
+    )
+
+    self.LYRICS_TERMS = [
+        "star wars",
+        "star trek",
+        "luke skywalker",
+        "captain kirk",
+        "spock",
+        "yoda",
+    ]
+    self.DANCEABILITY_RANGE = [0.45, 0.99]
+    self.ENERGY_RANGE = [0.45, 0.75]
+    self.TEMPO_RANGE = [110.0, 140.0]
+    self.MIN_VIEWS = 1000
+
+    self.provider.io.info(f"{self.name} initialized.")
+```
+
+> :information_source: To make your Input plugin more dynamic, many of the variables above can be made to be read in from the user using the [Alteryx UI-SDK](https://github.com/alteryx/dev-harness)!
+
+## 4. Data Processing
+Now it's time to process some data! Since we do not currently attempt incoming data, but rather, _provide_ data as output, we'll be doing the bulk of our work in `on_complete`.
```

Comment:
TW review:

Now it's time to process some data! Since we don't currently attempt incoming data, but rather, _provide_ data as output, we'll do the bulk of our work in `on_complete`.